pkg/executor/service.go

```diff
@@ -160,33 +163,48 @@ func UpdatePodService(
 		reqServiceType,
 		service.Name,
 	)
-	// serviceType has been updated. Create a patch to change the type
-	type patchSpec struct {
-		Op    string         `json:"op"`
-		Path  string         `json:"path"`
-		Value v1.ServiceType `json:"value"`
-	}
 
-	patch := []patchSpec{
-		{
-			Op:    "replace",
-			Path:  "/spec/type",
-			Value: reqServiceType,
+	// TODO: should we just use the service that was passed in?
```

Comment:
Similar comments here as for the other patch situation. In this case the service pointer originally comes from queryService (which, in the non-nil case, is returning the result of observer.GetService).

Reply:
More discussion of PATCH stuff in my long comment below.